src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/enumerations/SentenceType.kt

```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.enumerations
+
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.AFineSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.AbstractSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.BotusSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.DetentionAndTrainingOrderSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.ExtendedDeterminateSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.SopcSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.StandardDeterminateSentence
+
+enum class SentenceType(val sentenceClazz: Class<out AbstractSentence>? = null) {
+  StandardDeterminate(StandardDeterminateSentence::class.java),
+  ExtendedDeterminate(ExtendedDeterminateSentence::class.java),
+  Sopc(SopcSentence::class.java),
+  AFine(AFineSentence::class.java),
+  DetentionAndTrainingOrder(DetentionAndTrainingOrderSentence::class.java),
+  Botus(BotusSentence::class.java),
+  Indeterminate(),
+  Unsupported(),
```

Comment:
I think this is probably out of scope. But if we're thinking of this enum as being the future representation of sentence type. Then we wouldn't really have an unsupported type, we'd have more types here without a `sentenceClazz` which are therefore unsupported.

Reply:
I'm also not sure we should have `Unsupported` as a sentence type. Its again mixing CRDS into a general SentenceType. Could this be Other? or Unknown until we've refined the sentence types more like the above comment.

Reply:
Replacing with NULL as previous. That way we can slowly make things "known" i.e. not null